go.mod

```diff
@@ -74,15 +74,22 @@ require (
 	github.com/google/gofuzz v1.0.0
 	github.com/mark3labs/mcp-go v0.18.0
 	github.com/ncruces/go-sqlite3 v0.22.0
+	go.etcd.io/etcd/client/v3 v3.5.21
 	go.uber.org/mock v0.5.0
 )
 
 require (
+	github.com/coreos/go-semver v0.3.0 // indirect
+	github.com/coreos/go-systemd/v22 v22.3.2 // indirect
 	github.com/ncruces/julianday v1.0.0 // indirect
 	github.com/tetratelabs/wazero v1.8.2 // indirect
 	github.com/tidwall/pretty v1.2.1 // indirect
 	github.com/xrash/smetrics v0.0.0-20240521201337-686a1a2994c1 // indirect
 	github.com/yosida95/uritemplate/v3 v3.0.2 // indirect
+	go.etcd.io/etcd/api/v3 v3.5.21 // indirect
+	go.etcd.io/etcd/client/pkg/v3 v3.5.21 // indirect
```

Comment:
Can we avoid etcd as hard dependency? It could be a plugin (similar to archiver/gcloud). 
Folks in K8s world might not use etcd directly but there's k8s lease API which can be leveraged as a leader election library. See https://github.com/edgedelta/leader-election

Reply:
Good point. Reworking this into a registry with a module.
But I think on k8s you will still use etcd, rather than k8s abstraction. We need a place to store heartbeats/shards allocations, and the k8s interface won't help in that. But we can keep it as an opportunity.

Reply:
given how few other dependencies changed, I'm kinda assuming the etcd library is rather lightweight.  may be fine tbh, making a bunch of separate modules has moderate maintenance costs.

do let me know if I'm wrong about that tho, like it's actually a horrific beast under the hood or something.